### PR TITLE
Ks/#350 unsetting general options

### PR DESCRIPTION
**Pull request by KSchopmeyer**
Extends functionality by allowing the value "" to unset selected general options in the interactive mode. 

See commit message for details.

### CODE REVIEW

File: pywbemtools/pywbemcli/pywbemcli.py [hunk 1]
```diff
@@ -304,32 +326,36 @@ def _resolve_mock_server(mock_server):
...
                    format(ev=PywbemServer.defaultnamespace_envvar,
                           default=DEFAULT_NAMESPACE))
 @click.option('-o', '--output-format', metavar='FORMAT',
+              default=None,  # There is no default value
               help=u'Output format for the command result. '
                    u'The default and allowed output formats are command '
                    u'specific. '
                    u'The default output_format is None so that each command '
-                   u'selects its own default format. '
+                   u'selects its own default format. Use '' to set default '
```
Comment (andy-maier) [inline pywbemtools/pywbemcli/pywbemcli.py:343]: Use double quotes.

File: pywbemtools/pywbemcli/pywbemcli.py [hunk 2]
```diff
@@ -348,14 +374,16 @@ def _resolve_mock_server(mock_server):
...
+              default=None,  # default value set in cli function
               envvar=PywbemServer.connections_file_envvar,
               # Keep help text in sync with connections file definitions in
               # _connection_repository.py:
               help=u'Path name of the connections file to be used. '
                    u'Default: EnvVar {ev}, or "{cf}" in the user\'s home '
                    u'directory (as determined using Python\'s '
                    u'os.path.expanduser("~"). See there for details, '
-                   u'particularly for Windows).'.
+                   u'particularly for Windows). Use '' to set default '
```
Comment (andy-maier) [inline pywbemtools/pywbemcli/pywbemcli.py:385]: Use double quotes.

**Review by andy-maier — APPROVED** [review]

**Review by andy-maier — CHANGES_REQUESTED** [review]
See comments

File: pywbemtools/pywbemcli/pywbemcli.py (hunk 1 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/pywbemcli.py:343]: DONE

File: pywbemtools/pywbemcli/pywbemcli.py (hunk 2 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/pywbemcli.py:385]: DONE

**Comment by KSchopmeyer** [comment]
OTE FOR FUTURE: I tried to simplify all this by using the click option callback to set the correct values and simplify the main path of the cli function but in the repl mode, all of the callback methods are executed for each character entered on the console (an atrifact of preparing the completion strings) which causes confusion and actually can impose a real load on pywbemcli.

**Comment by KSchopmeyer** [comment]
Discussion closed.  We agree not to allow reset of --server, --name, --mock-server

**Review by andy-maier — APPROVED** [review]

**Pull request merged**